### PR TITLE
[57] [170] aggregateMessages for python update

### PR DESCRIPTION
**Pull request by tsivula**
This PR implements the updates discussed in PR #169 post merge.

Updates:
- Move AggregateMessages under graphframes.lib
- rename `msgToX` into `sendToX`
- change parameter ``assert`` checks into raise errors
- check for string type (Python 2 and 3) for params `sendToX`

### CODE REVIEW

File: python/graphframes/lib/aggregate_messages.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#
+
+from pyspark import SparkContext
+from pyspark.sql import DataFrame, functions as sqlfunctions
+
+
+def _java_api(jsc):
+    javaClassName = "org.graphframes.GraphFramePythonAPI"
+    return jsc._jvm.Thread.currentThread().getContextClassLoader().loadClass(javaClassName) \
+            .newInstance()
```
Comment (felixcheung) [inline python/graphframes/lib/aggregate_messages.py]: interesting.. this wasn't there in the earlier implementation?

**Comment by felixcheung** [comment]
@jkbradley for reviewing this and heads up before rolling another release.

File: python/graphframes/lib/aggregate_messages.py (hunk 1 above)
Comment (tsivula) [inline python/graphframes/lib/aggregate_messages.py]: This is a copy from https://github.com/graphframes/graphframes/pull/196/files/61c9ae6fc942a47d76735e040d274f6484fab08c#diff-e81e6b169c0aa35012a3263b2f31b330L32
... from the module this new module was detached from. Maybe we should just import it instead?
``from graphframes.graphframe import _java_api``

Comment (jkbradley) [inline python/graphframes/lib/aggregate_messages.py]: Importing sounds reasonable

**Comment by jkbradley** [comment]
@tsivula Thanks for the update!
@felixcheung This LGTM, so I'll go ahead and merge it.

File: python/graphframes/lib/aggregate_messages.py (hunk 1 above)
Comment (jkbradley) [inline python/graphframes/lib/aggregate_messages.py]: Whoops, just merged this.  We can do the import sometime later.  I'll go ahead and start preparing a release to include this and the indexing bug fix.

**Comment by tsivula** [comment]
Thanks for the reviewers

**Pull request merged**